### PR TITLE
add Logic Inference Dataset

### PR DESCRIPTION
**Pull request by kkie02**
Regarding https://github.com/LAION-AI/Open-Assistant/issues/261.

This is an re-produce of the dataset from LogicInference Dataset in paper: https://openreview.net/pdf?id=HAGeIS_Lcg9. I think it will helpful for improving logic inference ability of the model.

The github page of LogicInference Dataset: https://github.com/google-research/google-research/tree/master/logic_inference_dataset.

This dataset is aimed to offer more dataset for Open Assistant project, depending on their demands, there three columns: INSTRUCTION, RESPONSE, SOURCE.

The results in this dataset is a little different from which was introduced in the original paper:

1.For all three splits (IID/OOD/length), only IID is used. In the original paper, it seems that model can reach better performance with data generated by this split method.

2.In the original paper, there are two form of responses: LOGICINFERENCEb (with the answer at the beginning) and LOGICINFERENCEe (with the answer at the end). This dataset uses LOGICINFERENCEe, that means: for all questions, the model will first do logic inference, and give the final answer at the end.

3.The original paper, some parameters in generate_dataset.py are:

N_INFERENCE_PROBLEMS = 5000

N_VARIATIONS = 25

N_EXAMPLES = 200000

TRAIN_RATIO = 0.9

LENGTH_SPLIT_THRESHOLD = 4

RANDOM_SEED = 0

I choose some new parameters:

N_INFERENCE_PROBLEMS = 10000

N_VARIATIONS = 25

N_EXAMPLES = 55000

TRAIN_RATIO = 1

LENGTH_SPLIT_THRESHOLD = 4

RANDOM_SEED = 1111

The original script generated 4814 different inference problems and extended all those inference problems to around 200,000 Q-A pairs. My settings generated 5491 different inference problems and extended them to around 54,607 Instruction-Response pairs. I think for Open Assistant projects, maybe the number of different inference problems is more important, and generated many similar Instruction-Response pairs will only add training time and doesn't make much sense.

### CODE REVIEW

**Comment by olliestanley** [comment]
Hi! Before we can review this we will need you to delete the `.idea/` so that the PR no longer tries to add it to the repository

**Comment by kkie02** [comment]
> Hi! Before we can review this we will need you to delete the `.idea/` directory so that the PR no longer tries to add it to the repository

Hi, I think that you can check this now if you have time. I just bought a new laptop and uses Windows wsl there, so I met many problems but now I fixed all of them. On the other hand, I forked the files of https://github.com/google-research/google-research/tree/master/logic_inference_dataset at the beginning, but their code doesn't follow the flake8 format, so I only keep the generate_dataset.py now and make it fitting the flake8 format. I did some changes in this file to make it suitable for our OA project.

By the way, should I still need to delete .idea/ file now? I am not sure.

**Comment by olliestanley** [comment]
> > Hi! Before we can review this we will need you to delete the `.idea/` directory so that the PR no longer tries to add it to the repository
> 
> Hi, I think that you can check this now if you have time. I just bought a new laptop and uses Windows wsl there, so I met many problems but now I fixed all of them. On the other hand, I forked the files of https://github.com/google-research/google-research/tree/master/logic_inference_dataset at the beginning, but their code doesn't follow the flake8 format, so I only keep the generate_dataset.py now and make it fitting the flake8 format. I did some changes in this file to make it suitable for our OA project.
> 
> By the way, should I still need to delete .idea/ file now? I am not sure.

Thanks. Yes we would still need the `.idea/` directory to be removed, as it's related to your local IDE setup so we should avoid having it in the repository

File: .idea/.gitignore
Comment (andreaskoepf) [inline .idea/.gitignore:1]: Would it be possible to remove the .idea folder from this PR please?

**Comment by andreaskoepf** [comment]
I removed the .idea folder & resolved the conflict...

**Review by andreaskoepf — APPROVED** [review]
Interesting dataset, curious if it has a noticeable impact on logical reasoning capabilities.

**Review by olliestanley — APPROVED** [review]

**Pull request merged**